File: WebContent/movie.js
function getParameterByName(target) {
    // Get request URL
    let url = window.location.href;
    // Encode target parameter name to url encoding
    target = target.replace(/[\[\]]/g, "\\$&");

    // Ues regular expression to find matched parameter value
    let regex = new RegExp("[?&]" + target + "(=([^&#]*)|&|#|$)"),
        results = regex.exec(url);
    if (!results) return null;
    if (!results[2]) return '';

    // Return the decoded parameter value
    return decodeURIComponent(results[2].replace(/\+/g, " "));
}

function handleResult(resultData) {

    let indexElement = jQuery("#back_to_index");
    let params = resultData[0]["params"];
    indexElement.append('<p> <a href= "movie-list.html?' + params +  '">' + "Back to Movies" + '</a> </p>');


    let titleElement = jQuery("#m_title");
    titleElement.append("<p>" + resultData[0]["movie_title"] + "</p>");

    let infoElement = jQuery("#m_info");
    infoElement.append(
        "<p>Director: " + resultData[0]["movie_director"] + "</p>" +
        "<p>Year: " + resultData[0]["movie_year"] + "</p>" +
        "<p>Rating: " + resultData[0]["movie_rating"] + "</p>" +
        "<button class='btn btn-info'  onclick=\"addMovie(\'" + resultData[0]['movie_id'] + "\')\"> Add to Cart </button>"
    );


    let genresElement = jQuery("#genres_table_body");
    genresElement.append(
        "<tr><td>" + resultData[0]["movie_genres"] +"</td></tr>"
    );

    let starsElement = jQuery("#stars_table_body");
    let stars_split = resultData[0]["movie_stars"].split(",");
    //get ids for stars
    let star_ids_split = resultData[0]["star_ids"].split(",");
    let rowHTML = "";
    for (let i = 0; i < stars_split.length; i++) {
        rowHTML += "<tr><td>" +  '<a href="star.html?id=' + star_ids_split[i] + '">' +stars_split[i]  + '</a>' + "</td></tr>";
    }
    starsElement.append(rowHTML);
}

let movieId = getParameterByName('id');

jQuery.ajax({
    dataType: "json",  // Setting return data type
    method: "GET",// Setting request method
    url: "api/Movie?id=" + movieId, // Setting request url, which is mapped by StarsServlet in Stars.java
    success: (resultData) => handleResult(resultData) // Setting callback function to handle data returned successfully by the SingleStarServlet
});

File: app/src/main/java/edu/uci/ics/fabflixmobile/Movie.java
package edu.uci.ics.fabflixmobile;

import org.json.JSONException;
import org.json.JSONObject;
import java.util.ArrayList;

public class Movie {
    private final String name;
    private final String director;
    private String movie_id;
    private final short year;
    private ArrayList<String> genres= new ArrayList<String>();
    private ArrayList<String> stars = new ArrayList<String>();

    public Movie(String name, short year) {
        this.name = name;
        this.year = year;
        director = "";
    }


    public Movie(JSONObject jsonObject) throws JSONException {
        this.name = jsonObject.getString("movie_title");
        this.year = (short) jsonObject.getInt("movie_year");
        this.director = jsonObject.getString("movie_director");
        this.movie_id = jsonObject.getString("movie_id");
        String[] starsString = jsonObject.getString("movie_stars").split(",");
        String[] genresString = jsonObject.getString("movie_genres").split(",");
        int t = 0;
        for (String i : starsString) {
            stars.add(i);
        }
        for (String i : genresString) {
            genres.add(i);
        }
    }

    public String getName() {
        return name;
    }

    public short getYear() {
        return year;
    }

    public String getDirector() {
        return director;
    }

    public String getMovie_id() {
        return movie_id;
    }

    public String getGenres() {
        String out = "";
        if (genres.size() == 1) {
            out = genres.get(0);
        }
        else if (genres.size() == 2) {
            out = genres.get(0) + ", " + genres.get(1);
        }
        else {
            out = genres.get(0) + ", " + genres.get(1) + ", " + genres.get(2);
        }
        return out;
    }

    public String getStars() {
        String out = "";
        if (stars.size() == 1) {
            out = stars.get(0);
        }
        else if (stars.size() == 2) {
            out = stars.get(0) + ", " + stars.get(1);
        }
        else {
            out = stars.get(0) + ", " + stars.get(1) + ", " + stars.get(2);
        }
        return out;
    }
}